hudi-utilities/src/main/java/org/apache/hudi/utilities/streamer/SourceFormatAdapter.java

```diff
@@ -158,8 +160,9 @@ public InputBatch<JavaRDD<GenericRecord>> fetchNewDataInAvroFormat(Option<String
       }
       case ROW: {
         //we do the sanitizing here if enabled
-        InputBatch<Dataset<Row>> r = ((Source<Dataset<Row>>) source).fetchNext(lastCkptStr, sourceLimit);
-        return new InputBatch<>(Option.ofNullable(r.getBatch().map(
+        // InputBatch<Dataset<Row>> r = ((Source<Dataset<Row>>) source).fetchNext(lastCkptStr, sourceLimit);
+        List<InputBatch<Dataset<Row>>> rs = ((Source<Dataset<Row>>) source).fetchNextForParitialUpdate(lastCkptStr, sourceLimit);
```

Comment:
Can you keep fetchNext as the main API to be called. Processing PartialUpdate or fullUpdate needs to be encapsulated inside the Source implementation.